app/models/ApplicationRequest.scala

```diff
@@ -56,67 +66,122 @@ case class EORIDetails(
   postcode: String,
   country: String
 )
+object EORIDetails {
+  implicit val format: OFormat[EORIDetails] = Json.format[EORIDetails]
+}
 
 case class ContactDetails(
   name: String,
   email: String,
   phone: Option[String]
 )
+object ContactDetails {
+  implicit val format: OFormat[ContactDetails] = Json.format[ContactDetails]
+}
 
-/** determined via `_type` field, e.g. "_type" = "MethodOne" { "_type": "MethodOne",
-  * "saleBetweenRelatedParties": "blah blah", ...
-  */
 sealed trait RequestedMethod
 
 case class MethodOne(
   saleBetweenRelatedParties: Option[String],
   goodsRestrictions: Option[String],
   saleConditions: Option[String]
 ) extends RequestedMethod
+object MethodOne {
+  implicit val format: OFormat[MethodOne] = Json.format[MethodOne]
+}
 
 sealed trait IdenticalGoodsExplaination
-// again, "_type": "PreviouslyImportedGoods", etc
 case class PreviousIdenticalGoods(value: String) extends IdenticalGoodsExplaination
+object PreviousIdenticalGoods {
+  implicit val format: OFormat[PreviousIdenticalGoods] = Json.format[PreviousIdenticalGoods]
+}
 case class OtherUsersIdenticalGoods(value: String) extends IdenticalGoodsExplaination
+object OtherUsersIdenticalGoods {
+  implicit val format = Json.format[OtherUsersIdenticalGoods]
```

Comment:
I think this should be `Json.valueFormat[T]`

Reply:
Updated in: https://github.com/hmrc/advance-valuation-rulings-frontend/pull/118/commits/4c49bc26badb7dafbbbfd552a23c19f802b4dd87